apps/myaccount/src/components/shared/header.tsx

```diff
@@ -167,6 +177,56 @@ export const Header: FunctionComponent<HeaderPropsInterface> = (
         }
     }, [ profileDetails?.profileInfo ]);
 
+    /**
+     * Handles language switch action.
+
+     * @param language - Selected language.
+     */
+    const handleLanguageSwitch = (language: string): void => {
+        moment.locale(language ?? "en");
+        I18n.instance.changeLanguage(language).catch((error: string | Record<string, unknown>) => {
+            throw new LanguageChangeException(language, error);
+        });
+
+        const cookieSupportedLanguage: string = language.replace("-", "_");
+        const domain: string = ";domain=" + extractDomainFromHost();
+        const cookieExpiryTime: number = 30;
+        const expires: string = "; expires=" + new Date().setTime(cookieExpiryTime * 24 * 60 * 60 * 1000);
+        const cookieString: string = "ui_lang=" + (cookieSupportedLanguage || "") + expires + domain + "; path=/";
+
+        CookieStorageUtils.setItem(cookieString);
+    };
+
+    /**
+     * Extracts the domain from the hostname.
+     * If parsing fails, undefined will be returned.
+     *
+     * @returns current domain
+     */
+    const extractDomainFromHost = (): string => {
+        let domain: string = undefined;
+
+        /**
+         * Extract the domain from the hostname.
+         * Ex: If console.wso2-is.com is parsed, `wso2-is.com` will be set as the domain.
+         * If hostname has no periods, then entire hostname is taken as domain. Ex: localhost
+         */
+        try {
+            const hostnameTokens: string[] = window.location.hostname.split(".");
+
+            if (hostnameTokens.length == 1){
+                domain = hostnameTokens[0];
+            } else if (hostnameTokens.length > 1) {
+                domain = hostnameTokens.slice(hostnameTokens.length - 2, hostnameTokens.length).join(".");
+            }
+        } catch (e) {
```

Comment:
Why do we need to catch this if we are not doing anything?

Reply:
A catch block or a final block is required after a try block in JavaScript[1]. Therefore, we need to have a catch block even if we are not doing anything inside it.

In the above code segment, we should ideally log the error that is caught by the catch block. A front end logger implementation is tracked via https://github.com/wso2/product-is/issues/11650 as mentioned in the comment in the catch block.

[1] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch